x/incentives/keeper/distribute_test.go

```diff
@@ -868,3 +870,196 @@ func (s *KeeperTestSuite) TestGetPoolFromGaugeId() {
+	s.App.EpochsKeeper.AfterEpochEnd(s.Ctx, epochInfo.GetIdentifier(), 1)
+
+	// Validate the results
+	clPoolInternalGaugeId, err := s.App.PoolIncentivesKeeper.GetPoolGaugeId(s.Ctx, clPoolId1.GetId(), epochInfo.Duration)
+	s.Require().NoError(err)
+
+	s.ValidateDistributedGauge(clPoolExternalGaugeIdPool1, 1, externalGaugeCoins)
+	s.ValidateDistributedGauge(clPoolExternalGaugeIdPool2, 1, halfOfExternalGaugeCoins)
+
+	internalIncentiveClPool, err := s.App.ConcentratedLiquidityKeeper.GetAllIncentiveRecordsForPool(s.Ctx, clPoolId1.GetId())
+	s.Require().NoError(err)
+
+	externalIncentiveClPool, err := s.App.ConcentratedLiquidityKeeper.GetAllIncentiveRecordsForPool(s.Ctx, clPoolId2.GetId())
+	s.Require().NoError(err)
+
+	// check  internalIncentiveCLPool
+	s.Require().Equal(2, len(internalIncentiveClPool))
+	emissionRate := sdk.NewDecFromInt(sdk.NewInt(1_000_000)).QuoTruncate(sdk.NewDec(epochInfo.Duration.Milliseconds()).QuoInt(sdk.NewInt(1000)))
+	s.ValidateIncentiveRecord(clPoolId1.GetId(), externalGaugeDecCoins[0], emissionRate, internalIncentiveClPool[0])
+	s.ValidateIncentiveRecord(clPoolId1.GetId(), externalGaugeDecCoins[1], emissionRate, internalIncentiveClPool[1])
+
+	// check  externalIncentiveCLPool
+	s.Require().Equal(2, len(externalIncentiveClPool))
+	emissionRate = sdk.NewDecFromInt(sdk.NewInt(1_000_000)).QuoTruncate(sdk.NewDec(epochInfo.Duration.Milliseconds() * 2).QuoInt(sdk.NewInt(1000)))
+	s.ValidateIncentiveRecord(clPoolId2.GetId(), halfOfExternalGaugeDecCoins[0], emissionRate, externalIncentiveClPool[0])
+	s.ValidateIncentiveRecord(clPoolId2.GetId(), halfOfExternalGaugeDecCoins[1], emissionRate, externalIncentiveClPool[1])
+
+	// 6. Remove distribution records for internal incentives using HandleReplacePoolIncentivesProposal
+	s.IncentivizeInternalGauge(clPoolId1.GetId(), clPoolId2.GetId(), epochInfo.Duration, true)
+
+	// 7. let epoch 2 pass
+	// ******************** EPOCH 2 *********************
+	s.Ctx = s.Ctx.WithBlockTime(s.Ctx.BlockTime().Add(epochInfo.Duration))
+	s.App.EpochsKeeper.AfterEpochEnd(s.Ctx, epochInfo.Identifier, 2)
+
+	s.ValidateDistributedGauge(clPoolInternalGaugeId, 2, internalGaugeCoins)
+	s.ValidateDistributedGauge(clPoolExternalGaugeIdPool1, 2, externalGaugeCoins)
+	s.ValidateDistributedGauge(clPoolExternalGaugeIdPool2, 2, externalGaugeCoins)
+
+	internalIncentiveClPoolEpoch2, err := s.App.ConcentratedLiquidityKeeper.GetAllIncentiveRecordsForPool(s.Ctx, clPoolId1.GetId())
+	s.Require().NoError(err)
+
+	externalIncentiveClPoolEpoch2, err := s.App.ConcentratedLiquidityKeeper.GetAllIncentiveRecordsForPool(s.Ctx, clPoolId2.GetId())
+	s.Require().NoError(err)
+
+	s.Require().Equal(3, len(internalIncentiveClPoolEpoch2))
+	emissionRate = sdk.NewDecFromInt(sdk.NewInt(1_000_000)).QuoTruncate(sdk.NewDec(epochInfo.Duration.Milliseconds()).QuoInt(sdk.NewInt(1000)))
+	emissionRateForStake := sdk.NewDecFromInt(sdk.NewInt(750_000)).QuoTruncate(sdk.NewDec(epochInfo.Duration.Milliseconds()).QuoInt(sdk.NewInt(1000)))
```

Comment:
nit: can we differentiate these var names more?

Reply:
renamed to 
emissionRate => emissionRateForExternalTokens
emissionRateForStake => emissionRateForInternalTokens

Reply:
It looks like emissionRate is used for both internal and and external tokens. And same goes for emissionRateForStake. Unless I am misunderstanding the test

Reply:
Where exactly do you think this is happening? 

they serve 2 different purpose;
internalEmissionRate has 750k coins distributed per second 
externalEmissionRate has 1M coins distributed per second

Reply:
ooh is it the wording `internalIncentiveClPoolEpoch2` and `externalIncentiveClPoolEpoch2`?

Reply:
yes

Reply:
gotchu, roman brought up the same issue, i changed it 

discussion: https://github.com/osmosis-labs/osmosis/pull/5500#discussion_r1235428162